File: src/main/java/org/taktik/connector/business/domain/common/Document.java
package org.taktik.connector.business.domain.common;

import org.taktik.freehealth.middleware.domain.SigningValue;

import java.io.Serializable;

/**
 * Created by aduchate on 8/11/13, 15:53
 */
public class Document implements Serializable {
    private String title;

    private byte[] content;
    private String textContent;

    private String filename;
    private String mimeType;
    private SigningValue signing;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public byte[] getContent() {
        return content;
    }

    public void setContent(byte[] content) {
        this.content = content;
    }

    public String getTextContent() {
        return textContent;
    }

    public void setTextContent(String textContent) {
        this.textContent = textContent;
    }

    public String getFilename() {
        return filename;
    }

    public void setFilename(String filename) {
        this.filename = filename;
    }

    public String getMimeType() {
        return mimeType;
    }

    public void setMimeType(String mimeType) {
        this.mimeType = mimeType;
    }

    public SigningValue getSigning() {
        return signing;
    }

    public void setSigning(SigningValue signing) {
        this.signing = signing;
    }
}

File: src/main/kotlin/org/taktik/freehealth/middleware/dto/common/Document.kt
/*
 *
 * Copyright (C) 2018 Taktik SA
 *
 * This file is part of FreeHealthConnector.
 *
 * FreeHealthConnector is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation.
 *
 * FreeHealthConnector is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with FreeHealthConnector.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

package org.taktik.freehealth.middleware.dto.common

import org.taktik.freehealth.middleware.domain.SigningValue
import java.io.Serializable

/**
 * Created by aduchate on 8/11/13, 15:53
 */
class Document(
    val title: String? = null,
    val content: ByteArray? = null,
    val textContent: String? = null,
    val filename: String? = null,
    val mimeType: String? = null,
    val signing: SigningValue? = null
) : Serializable
